test/e2e/federated-service.go

```diff
@@ -106,58 +117,60 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 			}
 			framework.Logf("%d clusters are Ready", len(contexts))
 
-			clusterClientSets = make([]*release_1_3.Clientset, len(clusterList.Items))
-			for i, cluster := range clusterList.Items {
-				framework.Logf("Creating a clientset for the cluster %s", cluster.Name)
+			// clusters = make([]cluster, len(clusterList.Items))
+			clusters = map[string]*cluster{}
+			primaryClusterName = clusterList.Items[0].Name
```

Comment:
We use the last cluster as the primary cluster in our setup/teardown scripts - https://github.com/kubernetes/kubernetes/blob/master/hack/e2e-internal/e2e-up.sh#L34